### ISSUE TITLE
Suggestion to allow notes to print with script

### ISSUE THREAD

**Issue by ftolsson** [lmparppei/Beat#93]
I know you hate settings, but IMO it is often valuable to print notes with scripts in certain situations. 
Here's my suggestion how to implement that with hopefully the least effort possible:

* Allow setting or menu item for "print notes with script".
* If ON, `body` element gets `.notesPrintWithScript`class
* Notes are always served to printout/preview html, but CSS gets two new entries:
```
.note {
display: none;
}
```
   and (for instance)
```
.notesPrintWithScript .note {
display: unset;
color: grey;
background-color: light-grey;
}
```
* the remaining issue would then be the paginator -- which I know less about -- but I am assuming that the paginator currently skips any text that is marked as a note. So telling the paginator that "if notes are set to print, then count notes too" is probably the hardest part in this scenario.

(As a side note, a similar scheme for synopsis would not be entirely bad either. :)

Here's hoping you agree!

**Comment by lmparppei** [comment]
There are multiple caveats to this, which would require a lot of discussion. I'll look into this, but I'm pretty sure this is not possible without major changes to both the parser and paginator.

**Comment by ftolsson** [comment]
Okidoke. I was just so thrilled at the thought that one could let the CSS make most of the heavy lifting. But I'm well aware that the paginator and parser are two quite mysterious creatures that I'm only able to make guesses about...

(Also: should I be less enthusiastic about CSS? Are you still deliberately moving away from CSS as I know you've mentioned, or was that just a result of the previous, less tamed styling that you inherited...?)

**Comment by lmparppei** [comment]
> Okidoke. I was just so thrilled at the thought that one could let the CSS make most of the heavy lifting. But I'm well aware that the paginator and parser are two quite mysterious creatures that I'm only able to make guesses about...

Parser preprocesses lines for printing and pagination, which means removing everything that's invisible (including empty lines). It *might* be enough to pass the lines to paginator and handle them as actions. 

> (Also: should I be less enthusiastic about CSS? Are you still deliberately moving away from CSS as I know you've mentioned, or was that just a result of the previous, less tamed styling that you inherited...?)

If I'm able to use (this code snippet)[https://github.com/cribaudo/Print-All-WKWebView-Content] to reliably print out WKWebView content to PDF, CSS will still be relevant. I really do hope so.

**Comment by lmparppei** [comment]
In `1.92.9` there is now an internal option to print out notes. Unfortunately, it's *only* notes, and not synopsis markers or sections, because allowing them would required a lot of work and code rewrites and redesign.

On HTML level, notes are wrapped in `<span class='note'>[[note]]</span>`, so styling them should be quite easy.

I'll be testing how it works out before rolling out the option to actually print them.

**Comment by lmparppei** [comment]
Finally we're seeing some advancement regarding this. The new pagination and rendering system made stuff a bit more complicated, and I'm still struggling to find an elegant solution, but currently, section headings and synopsis markers can be printed out pretty reliably. I think `#` and `=` should be spared in front of them, though.

Next stop is notes, but it's a tad more complicated, because I've built the system completely around stripping away any invisible text.

<img width="666" alt="image" src="https://github.com/lmparppei/Beat/assets/20226704/8890995e-8585-4564-8d6c-821558b2ec31">

**Comment by lmparppei** [comment]
And notes are now supported as well.

<img width="669" alt="image" src="https://github.com/lmparppei/Beat/assets/20226704/615c11e1-6abf-425b-a7e5-b25bc0890c4a">

**Comment by lmparppei** [comment]
Implemented in **1.99.69** [1541c01](https://github.com/lmparppei/Beat/commit/1541c01e7d2e53ed74bf688936b25f5fc7442ce4) 

<img width="880" alt="image" src="https://github.com/lmparppei/Beat/assets/20226704/f755c51a-f1a1-41f0-9272-cfaca4e24e24">
